process/block/preprocess/transactions.go

```diff
@@ -220,65 +232,249 @@ func (txs *transactions) ProcessBlockTransactions(
 	haveTime func() bool,
 ) error {
 
-	mapHashesAndTxs := txs.GetAllCurrentUsedTxs()
-	expandedMiniBlocks, err := txs.miniBlocksCompacter.Expand(block.MiniBlockSlice(body), mapHashesAndTxs)
-	if err != nil {
-		return err
+	if txs.isBodyToMe(body) {
+		return txs.processTxsToMe(body, haveTime)
 	}
 
-	// basic validation already done in interceptors
-	for i := 0; i < len(expandedMiniBlocks); i++ {
-		miniBlock := expandedMiniBlocks[i]
+	if txs.isBodyFromMe(body) {
+		return txs.processTxsFromMe(body, haveTime)
+	}
+
+	return process.ErrInvalidBody
+}
+
+func (txs *transactions) computeTxsToMe(body block.Body) ([]*txcache.WrappedTransaction, error) {
+
+	allTxs := make([]*txcache.WrappedTransaction, 0, process.MaxItemsInBlock)
+
+	for _, miniBlock := range block.MiniBlockSlice(body) {
+		shouldSkipMiniblock := miniBlock.SenderShardID == txs.shardCoordinator.SelfId() || !txs.isMiniBlockCorrect(miniBlock.Type)
+		if shouldSkipMiniblock {
+			continue
+		}
 		if miniBlock.Type != txs.blockType {
```

Comment:
this check was already done with txs.isMiniBlockCorrect - there is no need for this check right now, and the log and error return as well.

Reply:
Actually this will check if a miniblock with destination in self shard has a type of InvalidBlock and will reject it. This situation is not caught in isMiniBlockCorrect method